src/main/java/com/test/fibonacci/controller/FibonacciController.java

```diff
@@ -0,0 +1,42 @@
+package com.test.fibonacci.controller;
+
+import com.test.fibonacci.businesslogic.entity.Response;
+import com.test.fibonacci.handler.FibonacciHandler;
+import com.test.fibonacci.util.UserMessages;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@Validated
+public class FibonacciController {
+
+    private FibonacciHandler handler;
+
+    @Autowired
+    public FibonacciController(FibonacciHandler handler) {
+        this.handler = handler;
+    }
+
+    @GetMapping("/get")
+    public Response getNextFibonacciNumber(@RequestParam("userId") @Min(0)
+            @NotNull(message = UserMessages.NULL_ERROR_MESSAGE) Long userId) {
+        return Response.builder().nextNumber(handler.handleGet(userId)).build();
+    }
+
+    @RequestMapping("/goBack")
+    public void goBack(@RequestParam("userId") @Min(0) @NotNull(message = UserMessages.NULL_ERROR_MESSAGE) Long userId) {
+        handler.handleGoingBack(userId);
+    }
+
+    @GetMapping("/list")
+    public Response listSequence(@RequestParam("userId") @Min(0) @NotNull(message = UserMessages.NULL_ERROR_MESSAGE) Long userId) {
+        return Response.builder().fibonacciSequence(handler.handleList(userId)).build();
+    }
```

Comment:
Conventia de denumire endpoint-uri nu e prea "restful". Teoretic un API REST, trebuie sa isi expuna functionalitatea sub forma unor resurse pe care se pot face citiri/scrieri/actualizari folosind verbe de genul CREATE/POST/UPDATE/DELETE.
In cazul de fata, eu cred ca as fi mers pe ceva de genul:

```
obtinere lista asociata utilizatorului:
GET    api/users/:userId:/fibonacci-sequence 
cu body gol 

creare element nou in lista asociata utilizatorului
POST   api/users/:userId:/fibonacci-sequence/elements 
cu body gol 

stergerea ultimului element din lista
DELETE  api/users/:userId:/fibonacci-sequence/elements 
cu body de genul `{ filterType: 'latest' } 
```
Pentru utimul endpoint, recunosc ca e loc de interpretare fiindca implica stergerea "ultimului element" fara ca acesta sa fie specificat si ca atare nu mi se pare ca se potriveste foarte bine o operatie de `DELETE` ce in principiu presupune de obicei specificarea resursei de sters folosind un identificator.